File: tests/williams_accumulation_distribution_test.py
import numpy as np
from taflow import WilliamsAccumulationDistribution
def test_williams_accumulation_distribution_lifecycle():
    s=WilliamsAccumulationDistribution(np.array([],float),np.array([],float),np.array([],float));s.extend([2],[1],[1.5]);assert s.value is not None;s.reset();assert len(s)==0

File: crates/taflow-core/src/indicators/williams_accumulation_distribution_test.rs
use super::williams_accumulation_distribution::WilliamsAccumulationDistribution;
#[test]
fn lifecycle() {
    let mut s = WilliamsAccumulationDistribution::new().unwrap();
    assert!(s.append(2.0, 1.0, 1.5).is_none());
    assert!(s.append(3.0, 1.5, 2.5).is_some());
    s.reset();
    assert!(s.value().is_none());
}
